cassovary-core/src/main/scala/com/twitter/cassovary/graph/UndirectedGraph.scala

```diff
@@ -0,0 +1,29 @@
+package com.twitter.cassovary.graph
+
+import com.twitter.cassovary.graph.StoredGraphDir.StoredGraphDir
+
+/*
+ * A graph where the outbound neighbors of each node equal the inbound neighbors.
+ * Given a DirectedGraph which stores the neighbors of each node (as outbound edges),
+ * this class wraps that graph in an undirected view.
+ */
+
+class UndirectedGraph(outboundGraph: DirectedGraph) extends DirectedGraph {
+  override def getNodeById(id: Int): Option[Node] = outboundGraph.getNodeById(id) map { new UndirectedNode(_)}
```

Comment:
That's not about this PR, but some thoughts related.

Is there any gain from this wrapper (UndirectedGraph) if getNodeById (and other methods) returns Node (normal, not an uniderected one). We cannot use UndirectedNode properties (unless casting).

Alternatively, we can have:
```
trait Graph[NodeType <: Node] {
  getNodeById: Option[NodeType]
}
trait UndirectedGraph extends Graph[UndirectedNode] {
  getNodeById: Option[UndirectedNode]
}
```